packages/common/src/ui/layout/tables/context/TableContext.ts

```diff
@@ -59,6 +61,52 @@ export const createTableStore = (): UseStore<TableStore> =>
         };
       });
     },
+    setRowStyle: (id, style) => {
+      set(state => ({
```

Comment:
what about extracting this to a method, and then using that for the two uses in `setRowStyles`?

```
const getStyledRowState = (
  rowState: Record<string, RowState>,
  id: string,
  style: AppSxProp | null
) => ({
  isSelected: false,
  isExpanded: false,
  isDisabled: false,
  ...rowState[id],
  style,
});
```

```
    setRowStyle: (id, style) => {
      set(state => ({
        ...state,
        rowState: {
          ...state.rowState,
          [id]: getStyledRowState(state.rowState, id, style),
        },
      }));
    },

// and.. 

Object.keys(rowState).forEach(id => (rowState[id] = getStyledRowState(rowState, id, null)));
ids.forEach(id => (rowState[id] = getStyledRowState(rowState, id, style)));
```

Reply:
Great idea! I took it a little further and made it a little generic and used it in *ALL* of the places.. oops df9e5c18ba391046441b1ee984282c5157ad5eec